Matrix.C

```diff
@@ -335,9 +337,9 @@ Matrix::pointwise_mult(
    const Vector& other,
    Vector& result) const
 {
-   CAROM_ASSERT(!result.distributed());
-   CAROM_ASSERT(!distributed());
-   CAROM_ASSERT(!other.distributed());
+   CAROM_VERIFY(!result.distributed());
+   CAROM_VERIFY(!distributed());
```

Comment:
In Ardra, there is an example of a distributed temporal basis matrix being pointwise multiplied with a non-distributed vector, and so setting this to VERIFY breaks the code. What do we think about only setting dimension checks to VERIFY (which is already done in the other merged PR) and leaving distributed checks as ASSERT?

Reply:
If the condition is `false`, with either `ASSERT` or `VERIFY`, then something is wrong. It sounds like the check itself is wrong, if it works in a case where the condition is `false`. That is, if a distributed matrix times a non-distributed vector works, then we should just remove the check entirely. In this case, if in Ardra you have a distributed matrix, and one of its rows is multiplied by a non-distributed vector, I'm not sure how that could work. The row index is local, so how is it specified in parallel? This function is supposed to multiply the vector by the row, which is stored entirely on the process owning that row, but that process would be the only one having the correct result and would need to do `MPI_Bcast` to broadcast the result to all processes. I'm guessing there is a peculiar usage that makes this work, but it doesn't sound right in general. Can you provide more details?

Reply:
@chldkdtn Perhaps Youngsoo can provide more accurate information on the meaning of the Ardra code and whether it can be rewritten to not break the ASSERT/VERIFY. 

Currently, the code is working, and for space-time ROM, there is an non-distributed initialized vector called carom_red_soln_init. It seems this vector can be distributed, but right now, it is not. The full order solution is projected onto the reduced order vector via: projectLiftedSolution(carom_red_soln_init, carom_soln, spatialbasis);. Then the reduced order solution is scaled by the temporal basis via: scaleWithTemporalBasis(carom_red_soln_init, temporalbasis, 0), where only row = 0 is chosen. This scaling function uses pointwise_mult and throws an error due to temporalbasis failing !distributed().

Reply:
So local row 0 on every process scales the same undistributed vector on every process, with different results on every process? Is it used on every process?

Reply:
Hi Dylan. You were right. It seems that temporal basis is distributed in name, but actually not distributed as all the processors have the same temporal basis. However, the spatial basis is distributed. And so, since they were created from one basis generator, both were set to distributed. I don't think libROM currently allows to set rightSV to not distributed while leftSV is.

Reply:
Looks like both of you have figured out what is going on with temporal basis. It is currently duplicated over the processors. If we can generalize the case, e.g., allowing distributed spatial basis and non-distributed temporal basis, that would be wonderful.

Reply:
It sounds like the temporal basis should be non-distributed, in which case no changes to the `Matrix` class would be needed (just changes in the basis generator). I am not yet familiar with this issue or a test that could verify a libROM change is working, so I propose just keeping this as an `ASSERT` in this PR and letting @kevinhkhuynh fix this for Ardra in another PR. Does that sound good @kevinhkhuynh @chldkdtn ?

Reply:
@dylan-copeland @kevinhkhuynh Please remind me again. ASSERT becomes only active with debug mode, while VERIFY is active both with optimized and debug modes, right?

Reply:
@dylan-copeland That sounds good to me.

@chldkdtn ASSERT is only active in debug mode, while VERIFY is active in all modes.